app/controllers/api/custom_buttons_controller.rb

```diff
@@ -0,0 +1,54 @@
+module Api
+  class CustomButtonsController < BaseController
+    def create_resource(_type, _id, data)
+      klass = collection_class(:custom_buttons)
+      data['resource_action'] = add_resource_action_resource(data['resource_action']) if data.key?('resource_action')
+      data['userid'] = User.current_user.userid
+      klass.create!(data.deep_symbolize_keys)
+    rescue => err
+      raise BadRequestError, "Failed to create new custom button - #{err}"
+    end
+
+    def edit_resource(type, id, data)
+      custom_button = fetch_custom_button(type, id)
+      updated_data = data['resource'] || data
+      updated_data['resource_action'] = add_resource_action_resource(updated_data['resource_action']) if updated_data.key?('resource_action')
+      custom_button.update_attributes!(updated_data.deep_symbolize_keys) if data.present?
+      custom_button
+    rescue => err
+      raise BadRequestError, "Failed to update custom button - #{err}"
+    end
+
+    def options
+      render_options(:custom_buttons, build_custom_button_options)
+    end
+
+    def build_custom_button_options
+      {
+        :custom_button_types => CustomButton::TYPES
+      }
+    end
+
+    private
+
+    def add_resource_action_resource(data)
```

Comment:
WDYT about calling this `find_or_create_resource_action` ?

Reply:
Also, did you consider adding `accepts_nested_attributes_for :resource_action` to `CustomButton`? I think that could be a good idea, since we don't expose `ResourceAction` in the API - this code is jumping through a lot of hoops just to extract the id from the request body (it shouldn't ever be an href, which is half of what `parse_id` does). Accepting nested attributes would make this controller incredibly simple 😁

Reply:
Ok, addressed both in 1d85410
The change above, required an alteration in the spec.